### DISCUSSION TITLE
BLE RSSI

### DISCUSSION THREAD

**Discussion by Droppix** [google/nearby#2449] [Q&A] [answered]
Hello,

First of all, many thanks for this product, even if there's still work to be done....

In a certain configuration, I'd like to use a BLE connection (Android / iOS) rather than Local Network to connect 2 devices, but within a given perimeter (i.e. < 1meter).
Is it possible with Nearby to somehow know the distance (estimate) between the 2 devices (Bluetooth RSSI)?

Thanks

**Comment by bourdakos1** [comment, marked as answer]
The short answer is no.

The C++ API has a callback `endpoint_distance_changed_cb` that will return very general distance info ("very close", "close", "far"). However, this isn't exposed in the Swift API or in Android. Additionally, the BLE medium isn't enabled for iOS yet, so using the C++ API would not actually provide updates on distance.

This most likely won't ever be exposed in Nearby Connections, but more likely to be handled as a part of Nearby Presence.

**Comment by anandgupadhyay** [comment]
@Droppix  were you able  to find this distance or strength of signal in iOS or Android?
